docker/ubuntu/kodeb/Dockerfile

```diff
@@ -0,0 +1,17 @@
+ARG REGISTRY=docker.io
+FROM $REGISTRY/koreader/kobase:0.3.2-20.04 AS base
+
+FROM base AS build
+
+USER root
+RUN apt-get update
+RUN apt-get upgrade -y
+ARG TOOLS="chrpath"
```

Comment:
Hm, but I notice chrpath is merely 41 kB. I figure it should either go in base (so that can be used for x86_64 all by itself) or dpkg-dev should migrate here?

Reply:
I have not looked at why it's even needed.

Reply:
I'm not sure this is still needed with https://github.com/koreader/koreader-base/pull/1638:
```bash
    # try to remove rpath
    if command_exists chrpath; then
        find "${BASE_DIR}/usr/lib/koreader/libs" -type f -name "*.so*" -print0 | xargs -0 chrpath -d
    else
        echo "chrpath tool not found. Skipping RPATH deletion"
    fi
```
Cf. https://github.com/koreader/koreader/pull/9351/files#diff-c5a4c21372a013bd579b179ad7c8dd0a2a14bc60a36ee5e78d57f275c16252a1R105.

Reply:
Deletion seems to imply something slightly different than setting -rpath $ORIGIN and -rpath $ORIGIN/libs but presumably the latter is actually better. Do you agree @pazos ?

Reply:
No idea. I used `chrpath` to remove a bunch of `lintian` warnings.

AFAIK current image doesn't have `chrpath` so you could run lintian against the last release and see if it raises some warnings.

Reply:
This is the full output:
```
E: koreader: embedded-library freetype [usr/lib/koreader/libs/libfreetype.so.6]
E: koreader: embedded-library libjpeg [usr/lib/koreader/libs/libjpeg.so.8]
E: koreader: embedded-library libjpeg [usr/lib/koreader/libs/libturbojpeg.so]
E: koreader: embedded-library libpng [usr/lib/koreader/libs/libpng16.so.16]
E: koreader: embedded-library libzstd [usr/lib/koreader/libs/libzstd.so.1]
E: koreader: embedded-library openjpeg [usr/lib/koreader/libs/libmupdf.so]
E: koreader: embedded-library zlib [usr/lib/koreader/libs/libz.so.1]
W: koreader: duplicate-font-file also in (fonts-freefont-ttf) [usr/lib/koreader/fonts/freefont/FreeSans.ttf]
W: koreader: duplicate-font-file also in (fonts-freefont-ttf) [usr/lib/koreader/fonts/freefont/FreeSerif.ttf]
W: koreader: national-encoding [usr/lib/koreader/frontend/ui/elements/font_ui_fallbacks.lua]
W: koreader: national-encoding [usr/lib/koreader/l10n/templates/mac.strings]
W: koreader: national-encoding [usr/lib/koreader/resources/icons/mdlight/LICENSE-SIL-OFL-1.1.md]
W: koreader: opentype-font-wrong-filename [usr/lib/koreader/fonts/freefont/FreeSans.ttf]
W: koreader: opentype-font-wrong-filename [usr/lib/koreader/fonts/freefont/FreeSerif.ttf]
W: koreader: script-not-executable [usr/lib/koreader/plugins/terminal.koplugin/profile]
W: koreader: wrong-name-for-changelog-of-native-package [usr/share/doc/koreader/changelog.Debian.gz]
```

Reply:
After merging to master (particularly base), and without chrpath installed:
```
lintian koreader-2024.04-12-arm64.deb
E: koreader: embedded-library usr/lib/koreader/libs/libfreetype.so.6: freetype
E: koreader: embedded-library usr/lib/koreader/libs/libjpeg.so.8: libjpeg
E: koreader: embedded-library usr/lib/koreader/libs/libmupdf.so: openjpeg
E: koreader: embedded-library ... use --no-tag-display-limit to see all (or pipe to a file/program)
W: koreader: duplicate-font-file usr/lib/koreader/fonts/freefont/FreeSans.ttf also in fonts-freefont-ttf
W: koreader: duplicate-font-file usr/lib/koreader/fonts/freefont/FreeSerif.ttf also in fonts-freefont-ttf
W: koreader: opentype-font-wrong-filename usr/lib/koreader/fonts/freefont/FreeSans.ttf
W: koreader: opentype-font-wrong-filename usr/lib/koreader/fonts/freefont/FreeSerif.ttf
W: koreader: script-not-executable usr/lib/koreader/plugins/terminal.koplugin/profile
```

Reply:
Really, stripping RPATH entries seem wrong.

Reply:
And on the subject of library paths, there's still some instances of `LD_LIBRARY_PATH` use:
```grep
frontend/apps/reader/modules/readerdictionary.lua:813:        -- NOTE: Bionic doesn't support rpath, but does honor LD_LIBRARY_PATH...
frontend/apps/reader/modules/readerdictionary.lua:816:            C.setenv("LD_LIBRARY_PATH", "./libs", 1)
frontend/apps/reader/modules/readerdictionary.lua:821:            C.unsetenv("LD_LIBRARY_PATH")
platform/kobo/nickel.sh:6:# NOTE: LD_LIBRARY_PATH is the only late export from rcS we don't siphon in koreader.sh, for obvious reasons ;).
platform/kobo/nickel.sh:7:export LD_LIBRARY_PATH="/usr/local/Kobo"
platform/pocketbook/koreader.app:9:export LD_LIBRARY_PATH="${KOREADER_DIR}/libs${LD_LIBRARY_PATH+:}${LD_LIBRARY_PATH}"
```

Apart from the kobo use-cases, I'm not sure they should remain.

Reply:
That's good. I've removed the RPATH delete hack from the debian script.

Reply:
> Apart from the kobo use-cases, I'm not sure they should remain.

The Bionic one sounds most crucial of all unless I'm overlooking something?

Reply:
It is, yes.